diff --git a/cfe/fsw/cfe-core/src/sb/cfe_sb_init.c b/cfe/fsw/cfe-core/src/sb/cfe_sb_init.c
index f3309c2..460595d 100644

# Copyright (c) 2018 NSF Center for Space, High-performance, and Resilient Computing (SHREC)
# University of Pittsburgh. All rights reserved.

# Redistribution and use in source and binary forms, with or without modification, are permitted provided
# that the following conditions are met:
# 1. Redistributions of source code must retain the above copyright notice,
#    this list of conditions and the following disclaimer.
# 2. Redistributions in binary form must reproduce the above copyright notice,
#    this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
# 
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. 
# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
# OF SUCH DAMAGE.
--- a/cfe/fsw/cfe-core/src/sb/cfe_sb_init.c
+++ b/cfe/fsw/cfe-core/src/sb/cfe_sb_init.c
@@ -158,7 +158,12 @@ int32 CFE_SB_EarlyInit (void) {
 
    CFE_SB.ZeroCopyTail = NULL;
 
-    return Stat;
+   Stat = CFE_SB_DDSInit();
+   if (Stat != CFE_SUCCESS) {
+       return Stat;
+   }
+
+   return Stat;
 
 }/* end CFE_SB_EarlyInit */

diff --git a/cfe/fsw/cfe-core/src/sb/cfe_sb_priv.h b/cfe/fsw/cfe-core/src/sb/cfe_sb_priv.h
index 460595d..dc40d82 100644
--- a/cfe/fsw/cfe-core/src/sb/cfe_sb_priv.h
+++ b/cfe/fsw/cfe-core/src/sb/cfe_sb_priv.h
@@ -400,6 +400,7 @@ CFE_SB_DestinationD_t *CFE_SB_GetDestinationBlk(void);
 int32 CFE_SB_PutDestinationBlk(CFE_SB_DestinationD_t *Dest);
 int32 CFE_SB_AddDest(uint16 RtgTblIdx, CFE_SB_DestinationD_t *Dest);
 int32 CFE_SB_RemoveDest(uint16 RtgTblIdx, CFE_SB_DestinationD_t *Dest);
+int CFE_SB_DDSInit(void);
 
 /*
  * External variables private to the software bus module
