diff --git a/cfe/fsw/cfe-core/src/make/make-rules.mak b/cfe/fsw/cfe-core/src/make/make-rules.mak
index 5babe78..f37faca 100644

# Copyright (c) 2018 NSF Center for Space, High-performance, and Resilient Computing (SHREC)
# University of Pittsburgh. All rights reserved.

# Redistribution and use in source and binary forms, with or without modification, are permitted provided
# that the following conditions are met:
# 1. Redistributions of source code must retain the above copyright notice,
#    this list of conditions and the following disclaimer.
# 2. Redistributions in binary form must reproduce the above copyright notice,
#    this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
# 
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. 
# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
# OF SUCH DAMAGE.

--- a/cfe/fsw/cfe-core/src/make/make-rules.mak
+++ b/cfe/fsw/cfe-core/src/make/make-rules.mak
@@ -16,6 +16,9 @@ default::$(TARGET)
 .c.o:
 	$(COMPILER) $(LOCAL_COPTS) $(INCLUDE_PATH) $(COPTS) $(DEBUG_OPTS) -DOS_DEBUG_LEVEL=$(DEBUG_LEVEL) -c -o $@ $<
 
+.cpp.o:
+	$(COMPILER) $(LOCAL_COPTS) $(INCLUDE_PATH) $(COPTS) $(DEBUG_OPTS) -DOS_DEBUG_LEVEL=$(DEBUG_LEVEL) -c -o $@ $<
+
 ## 
 ## Assembly Code Rule
 ##
diff --git a/cfe/fsw/cfe-core/src/sb/cfe_sb.mak b/cfe/fsw/cfe-core/src/sb/cfe_sb.mak
index b283a99..d51e61c 100644
--- a/cfe/fsw/cfe-core/src/sb/cfe_sb.mak
+++ b/cfe/fsw/cfe-core/src/sb/cfe_sb.mak
@@ -7,18 +7,29 @@
 # History:
 ###############################################################################
 
+# OpenSplice dependencies.
+LOCAL_COPTS = -fPIC -O3 -DNDEBUG -D_REENTRANT -D_POSIX_C_SOURCE=200112L -D_XOPEN_SOURCE=500 -Wall -Wno-unused-function -I$(OSPL_HOME) -I$(OSPL_HOME)/dds -I$(OSPL_HOME)/include/dcps/C++/SACPP \
+ -I$(OSPL_HOME)/include -I$(OSPL_HOME)/include/sys -L$(OSPL_HOME)/dds -L$(OSPL_HOME)/lib \
+ -std=c++11
+
 # Subsystem produced by this makefile.
 TARGET = sb.o
 
 #==============================================================================
 # Object files required to build subsystem.
 
-OBJS=\
- cfe_sb_task.o cfe_sb_api.o cfe_sb_priv.o cfe_sb_init.o cfe_sb_buf.o \
- cfe_sb_util.o ccsds.o
+C_OBJS=\
+ cfe_sb_task.o cfe_sb_api.o cfe_sb_api_mod.o cfe_sb_priv.o cfe_sb_init.o \
+ cfe_sb_buf.o cfe_sb_util.o ccsds.o
+
+CPP_OBJS = \
+ cfe_sb_api_dds.o SBDcps.o SBDcps_impl.o SB.o SBSplDcps.o\
+ DataReaderListenerImpl.o
+
+OBJS = $(C_OBJS) $(CPP_OBJS)
 
 #==============================================================================
 # Source files required to build subsystem; used to generate dependencies.
 
-SOURCES = $(OBJS:.o=.c)
+SOURCES = $(C_OBJS:.o=.c) $(CPP_OBJS:.o=.cpp)
 
